installers/rpm-code/codes/flash.sh

```diff
@@ -3,54 +3,51 @@
 flash_main() {
     codes_dependencies hypre
     codes_download_proprietary "flash/FLASH-4.6.2.tar.gz" "FLASH4.6.2"
-    local flash_src_root=$PWD
-    patch_makefile
-    for n in CapLaser RTFlame
+    flash_patch_makefile
+    install -d 755 "${codes_dir[share]}"/flash4
+    for n in CapLaserBELLA RTFlame
     do
-        setup_$n $n
-        make_and_install $n
+        flash_make_and_install_type $n
     done
 }
 
-make_and_install() {
-    cd $1
+flash_make_and_install_type() {
+    "flash_setup_$n" "$n"
+    cd "$1"
     codes_make
-    install -m 755 flash4 "${codes_dir[bin]}"/flash4-$1
-    install -d 755 "${codes_dir[share]}"/flash4
-    install -m 444 setup_units "${codes_dir[share]}"/flash4/setup_units-$1
+    # POSIT: Sirepo assumes the exe is named flash4-flashType
+    install -m 755 flash4 "${codes_dir[bin]}"/flash4-"$1"
+    # POSIT: Sirepo assumes the setup_units file is named setup_units-flashType
+    install -m 444 setup_units "${codes_dir[share]}"/flash4/setup_units-"$1"
     cd ..
 }
 
-patch_makefile() {
-    patch sites/Prototypes/Linux/Makefile.h <<'EOF'
+flash_patch_makefile() {
+    local d=$(dirname "$BIVIO_MPI_LIB")
+    patch sites/Prototypes/Linux/Makefile.h <<EOF
 6c6
 < MPI_PATH   = /usr/local/mpich2/
 ---
-> MPI_PATH   = /usr/lib64/mpich/
+> MPI_PATH   = $d/
 8,9c8,9
 < HDF5_PATH  = /usr/local/hdf5
 < HYPRE_PATH = /usr/local/hypre
 ---
-> HDF5_PATH  = /usr/lib64/mpich
-> HYPRE_PATH = /home/vagrant/.local
-157c157
-< #----------------------------------------------------------------------------
----
-> #----------------------------------------------------------------------------
+> HDF5_PATH  = $d/
+> HYPRE_PATH = ${codes_dir[prefix]}
 EOF
 }
 
-setup_CapLaser() {
-    cd source/Simulation/SimulationMain/magnetoHD
-    codes_download_proprietary "flash/CapLaserBELLA-4.6.2.tar.gz" "CapLaserBELLA"
-    cd $flash_src_root
-    ./setup -auto magnetoHD/CapLaser -2d -nxb=16 -nyb=16 +hdf5typeio \
+flash_setup_CapLaserBELLA() {
+    codes_download_proprietary "flash/$1-4.6.2.tar.gz" "source"
+    cd ..
+    ./setup "$1" -objdir="$1" -auto -2d -nxb=16 -nyb=16 +hdf5typeio \
             species=fill,wall +mtmmmt +usm3t +mgd mgd_meshgroups=6 \
-            -parfile=caplaser_basic.par -objdir=$1 +laser \
-            ed_maxPulses=1 ed_maxPulseSections=4 ed_maxBeams=1
+            -parfile=caplaser_basic.par +laser \ ed_maxPulses=1
```

Comment:
this line seems incorrect. The next line only works, because it is correct shell syntax

Reply:
Good catch. Emacs fill-region doesn't work here. I guess it probably can't. Too hard to parse.

Reply:
https://github.com/radiasoft/download/pull/95/commits/4f394e009f65fb6649b1f4a0d9103ddd9be63bde